pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -0,0 +1,199 @@
+	componentExt *v1beta1.ComponentExtensionSpec,
+	podSpec *corev1.PodSpec,
+	componentStatus v1beta1.ComponentStatusSpec) *appsv1.Deployment {
+	var minReplicas int32
+	if componentExt.MinReplicas == nil {
+		minReplicas = int32(constants.DefaultMinReplicas)
+	} else {
+		minReplicas = int32(*componentExt.MinReplicas)
+	}
+
+	if minReplicas < int32(constants.DefaultMinReplicas) {
+		minReplicas = int32(constants.DefaultMinReplicas)
+	}
+	setDefaultPodSpec(podSpec)
+	deployment := &appsv1.Deployment{
+		ObjectMeta: componentMeta,
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &minReplicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": constants.GetRawServiceLabel(componentMeta.Name, componentMeta.Namespace),
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": constants.GetRawServiceLabel(componentMeta.Name, componentMeta.Namespace),
+					},
+				},
+				Spec: corev1.PodSpec(*podSpec),
+			},
+		},
+	}
+	setDefaultDeploymentSpec(&deployment.Spec)
+	return deployment
+}
+
+//checkNativeDeploymentExist checks is the deployment is existed?
+func (r *DeploymentReconciler) checkDeploymentExist(client client.Client) (constants.CheckResultType, error) {
+	//get deployment
+	existingDeployment := &appsv1.Deployment{}
+	err := client.Get(context.TODO(), types.NamespacedName{
+		Namespace: r.Deployment.Namespace,
+		Name:      r.Deployment.Name,
+	}, existingDeployment)
+	if err != nil {
+		if apierr.IsNotFound(err) {
+			return constants.CheckResultCreate, nil
+		}
+		return constants.CheckResultUnknown, err
+	}
+	//existed, check equalment
+	if semanticDeploymentEquals(r.Deployment, existingDeployment) {
+		return constants.CheckResultExisted, nil
+	}
+	return constants.CheckResultUpdate, nil
+}
+
+func semanticDeploymentEquals(desired, existing *appsv1.Deployment) bool {
+	return equality.Semantic.DeepEqual(desired.Spec, existing.Spec)
+}
+
+func setDefaultPodSpec(podSpec *corev1.PodSpec) {
+	if podSpec.DNSPolicy == "" {
+		podSpec.DNSPolicy = corev1.DNSClusterFirst
+	}
+	if podSpec.RestartPolicy == "" {
+		podSpec.RestartPolicy = corev1.RestartPolicyAlways
+	}
+	if podSpec.TerminationGracePeriodSeconds == nil {
+		TerminationGracePeriodSeconds := int64(corev1.DefaultTerminationGracePeriodSeconds)
+		podSpec.TerminationGracePeriodSeconds = &TerminationGracePeriodSeconds
+	}
+	if podSpec.SecurityContext == nil {
+		podSpec.SecurityContext = &corev1.PodSecurityContext{}
+	}
+	if podSpec.SchedulerName == "" {
+		podSpec.SchedulerName = corev1.DefaultSchedulerName
+	}
+	for i, _ := range podSpec.Containers {
+		container := &podSpec.Containers[i]
+		if container.TerminationMessagePath == "" {
+			container.TerminationMessagePath = "/dev/termination-log"
+		}
+		if container.TerminationMessagePolicy == "" {
+			container.TerminationMessagePolicy = corev1.TerminationMessageReadFile
```

Comment:
Can we default to `FallbackToLogsOnError` to the error gets propagated back to inference service status